docs/modules/ROOT/pages/api/introspection.adoc

```diff
@@ -69,8 +69,8 @@ SRC5 contract implementation extending xref:ISRC5[`ISRC5`].
 * xref:#SRC5-deregister_interface[`++deregister_interface(self, interface_id)++`]
 --
 
-[#SRC5-External-Functions]
-==== External Functions
+[#SRC5-Embeddable-Implementations-Functions]
+==== Embeddable Implementations Functions
```

Comment:
Should we change the `self` type in the functions below to `ComponentState<TContractState>`?

Reply:
What do you think about leaving just ComponentState? I think the generic TState is implicit since TContractState is always required, and this will appear frequently in the APIs.

Reply:
Good call, and it's much less noisy. +1

Reply:
Actually, I think we should leave just `ContractState`, because the embeddable implementation generated when you use `embeddable_as` is the one having just ContractState, like this:

```
#[starknet::embeddable]
impl SRC5Impl<
            TContractState, +HasComponent<TContractState>
, impl TContractStateDrop: Drop<TContractState>
> of interface::ISRC5<TContractState> {
    
    fn supports_interface(self: @TContractState, interface_id: felt252
) -> bool {
        let component = HasComponent::get_component(self);
        SRC5::supports_interface(component, interface_id, )
    }
```

Reply:
The impl we embed is not the one we write in the component, but the one the compiler generates from the `embeddable_as` attribute. And this is the one the user would usually use. Is different for InternalImpl though, because this doesn't use `embeddable_as`. For this one I think we should use ComponentState.

Reply:
Hmm if we go that route, should we include a NOTE or something to explain the rationale behind using `ContractState` for embeddables and `ComponentState` for internals? IMO it'd be weird to not say anything about it

Reply:
I agree it can be confusing, but where can we add such note? It seems we would need to add it to every component API

Reply:
Repeating the same note is definitely not ideal. Even though this is about reading the API, it's IMO more about understanding the component architecture (as you described above). Maybe this can be addressed in the `Extensibility` doc

Reply:
Agree, let's stick with ContractState for embeddable impls functions, and ComponentState for InternalImpl in components, at least until we revisit how we present this in the Extensibility PR. wdyt?

Reply:
Sounds good to me